### ISSUE TITLE
Post upgrade, reset & sync only moving maybe 1% of rows

### ISSUE THREAD

**Issue by qops1981** [airbytehq/airbyte#11257]
## Environment
- **Airbyte version**: `0.35.17-alpha` .. `0.35.55-alpha`
- **OS Version / Instance**: AWS EC2
- **Deployment**: Docker
- **Source Connector and version**: Postgres `0.4.9`
- **Destination Connector and version**: Snowflake `0.4.20` w/S3 Staging
- **Severity**: ~Critical~ Low
- **Step where error happened**: Sync job

## Current Behavior
After Pulling 110's of Millions of rows, only a couple thousand make it into Snowflake

## Expected Behavior
All rows pulled from Postgres should make it into Snowflake

## Additional Contest
We were at AirByte `0.35.13` I updated us to `0.35.55` as well as all the connectors for Postgres, S3, & Snowflake. During this update I was also consolidating the connections we add so there were fewer with more tables. After resetting data and Syncing we found there were far far fewer rows on the Snowflake side. Looking in the logs we can see after finishing the pull from Postgres with millions of rows, there are some Thread errors and it calls out only flushing 8k rows, which is how many I found in Snowflake.  This has halted our analytics and reporting until we can figure out how to proceed.

I have tested downgrading airbyte down to `0.35.17` but not the connectors and there was no success there. I am currently testing a downgrade of the Snowflake connector from `0.4.20` to `0.4.7` to see if that's the cause. But this will take 8 hours to run. Hoping maybe the community may have better insight. I cut only a segment of what I found important in the logs below for review. If you want the whole file, let me know.  

## Logs
*If applicable, please upload the logs from the failing operation. 
For sync jobs, you can download the full logs from the UI by going to the sync attempt page and 
clicking the download logs button at the top right of the logs display window.*

<details>
<summary>LOG</summary> 

```

2022-03-18 04:14:48 [32mINFO[m i.a.w.DefaultReplicationWorker(lambda$getReplicationRunnable$5):281 - Records read: 224141000
2022-03-18 04:14:48 [32mINFO[m i.a.w.DefaultReplicationWorker(lambda$getReplicationRunnable$5):281 - Records read: 224142000
2022-03-18 04:14:48 [32mINFO[m i.a.w.DefaultReplicationWorker(lambda$getReplicationRunnable$5):281 - Records read: 224143000
2022-03-18 04:14:48 [32mINFO[m i.a.w.DefaultReplicationWorker(lambda$getReplicationRunnable$5):281 - Records read: 224144000
2022-03-18 04:14:48 [32mINFO[m i.a.w.DefaultReplicationWorker(lambda$getReplicationRunnable$5):281 - Records read: 224145000
2022-03-18 04:14:48 [32mINFO[m i.a.w.DefaultReplicationWorker(lambda$getReplicationRunnable$5):281 - Records read: 224146000
2022-03-18 04:14:49 [44msource[0m > 2022-03-18 04:14:48 [32mINFO[m i.a.i.s.r.AbstractDbSource(lambda$createReadIterator$7):250 - Reading stream contacts. Records read: 110940000
2022-03-18 04:14:49 [32mINFO[m i.a.w.DefaultReplicationWorker(lambda$getReplicationRunnable$5):281 - Records read: 224147000
2022-03-18 04:14:49 [32mINFO[m i.a.w.DefaultReplicationWorker(lambda$getReplicationRunnable$5):281 - Records read: 224148000
2022-03-18 04:14:49 [32mINFO[m i.a.w.DefaultReplicationWorker(lambda$getReplicationRunnable$5):281 - Records read: 224149000
2022-03-18 04:14:49 [32mINFO[m i.a.w.DefaultReplicationWorker(lambda$getReplicationRunnable$5):281 - Records read: 224150000
2022-03-18 04:14:49 [43mdestination[0m > 2022-03-18 04:14:49 [32mINFO[m i.a.i.d.b.BufferedStreamConsumer(flushQueueToDestination):181 - Flushing buffer: 26213776 bytes
2022-03-18 04:14:49 [43mdestination[0m > 2022-03-18 04:14:49 [32mINFO[m i.a.i.d.b.BufferedStreamConsumer(lambda$flushQueueToDestination$1):185 - Flushing contacts: 25192 records
2022-03-18 04:14:49 [32mINFO[m i.a.w.DefaultReplicationWorker(lambda$getReplicationRunnable$5):281 - Records read: 224151000
2022-03-18 04:14:49 [32mINFO[m i.a.w.DefaultReplicationWorker(lambda$getReplicationRunnable$5):281 - Records read: 224152000
2022-03-18 04:14:49 [32mINFO[m i.a.w.DefaultReplicationWorker(lambda$getReplicationRunnable$5):281 - Records read: 224153000
2022-03-18 04:14:49 [43mdestination[0m > 2022-03-18 04:14:49 [32mINFO[m a.m.s.StreamTransferManager(uploadStreamPart):558 - [Manager uploading to textus-airbyte-workspace/public/contacts/2022_03_17_1647550729493_b7f89d35-e7d2-4bcf-8eb5-654935547193.csv with id VcBUJLp8S...x9pnk34iW]: Finished uploading [Part number 3 containing 10.01 MB]
2022-03-18 04:14:49 [32mINFO[m i.a.w.DefaultReplicationWorker(lambda$getReplicationRunnable$5):281 - Records read: 224154000
2022-03-18 04:14:50 [32mINFO[m i.a.w.DefaultReplicationWorker(lambda$getReplicationRunnable$5):281 - Records read: 224155000
2022-03-18 04:14:50 [32mINFO[m i.a.w.DefaultReplicationWorker(lambda$getReplicationRunnable$5):281 - Records read: 224156000
2022-03-18 04:14:50 [44msource[0m > 2022-03-18 04:14:49 [32mINFO[m i.a.i.s.r.AbstractDbSource(lambda$createReadIterator$7):250 - Reading stream contacts. Records read: 110950000
2022-03-18 04:14:50 [32mINFO[m i.a.w.DefaultReplicationWorker(lambda$getReplicationRunnable$5):281 - Records read: 224157000
2022-03-18 04:14:50 [32mINFO[m i.a.w.DefaultReplicationWorker(lambda$getReplicationRunnable$5):281 - Records read: 224158000
2022-03-18 04:14:50 [32mINFO[m i.a.w.DefaultReplicationWorker(lambda$getReplicationRunnable$5):281 - Records read: 224159000
2022-03-18 04:14:50 [44msource[0m > 2022-03-18 04:14:49 [32mINFO[m i.a.i.s.r.StateDecoratingIterator(computeNext):60 - State Report: stream name: AirbyteStreamNameNamespacePair{name='contacts', namespace='public'}, original cursor field: null, original cursor null, cursor field: updated_at, new cursor: 2022-03-18T00:39:57.778507Z
2022-03-18 04:14:50 [44msource[0m > 2022-03-18 04:14:49 [32mINFO[m i.a.i.s.r.AbstractDbSource(lambda$read$2):124 - Closing database connection pool.
2022-03-18 04:14:50 [44msource[0m > 2022-03-18 04:14:49 [32mINFO[m i.a.i.s.r.AbstractDbSource(lambda$read$2):126 - Closed database connection pool.
2022-03-18 04:14:50 [44msource[0m > 2022-03-18 04:14:49 [32mINFO[m i.a.i.b.IntegrationRunner(runInternal):154 - Completed integration: io.airbyte.integrations.base.ssh.SshWrappedSource
2022-03-18 04:14:50 [44msource[0m > 2022-03-18 04:14:49 [32mINFO[m i.a.i.s.p.PostgresSource(main):363 - completed source: class io.airbyte.integrations.source.postgres.PostgresSource
2022-03-18 04:14:50 [32mINFO[m i.a.w.DefaultReplicationWorker(run):152 - One of source or destination thread complete. Waiting on the other.
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [33mWARN[m i.a.i.b.IntegrationRunner(watchForOrphanThreads):231 - The main thread is exiting while children non-daemon threads from a connector are still active.
2022-03-18 04:14:50 [43mdestination[0m > Ideally, this situation should not happen...
2022-03-18 04:14:50 [43mdestination[0m > Please check with maintainers if the connector or library code should safely clean up its threads before quitting instead.
2022-03-18 04:14:50 [43mdestination[0m > The main thread is: main (RUNNABLE)
2022-03-18 04:14:50 [43mdestination[0m >  Thread stacktrace: java.base/java.lang.Thread.getStackTrace(Thread.java:1610)
2022-03-18 04:14:50 [43mdestination[0m >         at io.airbyte.integrations.base.IntegrationRunner.dumpThread(IntegrationRunner.java:268)
2022-03-18 04:14:50 [43mdestination[0m >         at io.airbyte.integrations.base.IntegrationRunner.watchForOrphanThreads(IntegrationRunner.java:235)
2022-03-18 04:14:50 [43mdestination[0m >         at io.airbyte.integrations.base.IntegrationRunner.runConsumer(IntegrationRunner.java:190)
2022-03-18 04:14:50 [43mdestination[0m >         at io.airbyte.integrations.base.IntegrationRunner.lambda$runInternal$1(IntegrationRunner.java:163)
2022-03-18 04:14:50 [43mdestination[0m >         at io.airbyte.integrations.base.sentry.AirbyteSentry.executeWithTracing(AirbyteSentry.java:54)
2022-03-18 04:14:50 [43mdestination[0m >         at io.airbyte.integrations.base.sentry.AirbyteSentry.executeWithTracing(AirbyteSentry.java:38)
2022-03-18 04:14:50 [43mdestination[0m >         at io.airbyte.integrations.base.IntegrationRunner.runInternal(IntegrationRunner.java:163)
2022-03-18 04:14:50 [43mdestination[0m >         at io.airbyte.integrations.base.IntegrationRunner.run(IntegrationRunner.java:105)
2022-03-18 04:14:50 [43mdestination[0m >         at io.airbyte.integrations.destination.snowflake.SnowflakeDestination.main(SnowflakeDestination.java:26)
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [33mWARN[m i.a.i.b.IntegrationRunner(watchForOrphanThreads):245 - Active non-daemon thread: pool-1121-thread-1 (WAITING)
2022-03-18 04:14:50 [43mdestination[0m >  Thread stacktrace: java.base@17.0.1/jdk.internal.misc.Unsafe.park(Native Method)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.locks.LockSupport.park(LockSupport.java:341)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionNode.block(AbstractQueuedSynchronizer.java:506)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ForkJoinPool.unmanagedBlock(ForkJoinPool.java:3463)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:3434)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:1623)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ArrayBlockingQueue.take(ArrayBlockingQueue.java:420)
2022-03-18 04:14:50 [43mdestination[0m >         at app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:484)
2022-03-18 04:14:50 [43mdestination[0m >         at app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:474)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.lang.Thread.run(Thread.java:833)
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [33mWARN[m i.a.i.b.IntegrationRunner(watchForOrphanThreads):245 - Active non-daemon thread: pool-1121-thread-2 (BLOCKED)
2022-03-18 04:14:50 [43mdestination[0m >  Thread stacktrace: app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:482)
2022-03-18 04:14:50 [43mdestination[0m >         at app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:474)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.lang.Thread.run(Thread.java:833)
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [33mWARN[m i.a.i.b.IntegrationRunner(watchForOrphanThreads):245 - Active non-daemon thread: pool-1121-thread-3 (BLOCKED)
2022-03-18 04:14:50 [43mdestination[0m >  Thread stacktrace: app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:482)
2022-03-18 04:14:50 [43mdestination[0m >         at app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:474)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.lang.Thread.run(Thread.java:833)
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [33mWARN[m i.a.i.b.IntegrationRunner(watchForOrphanThreads):245 - Active non-daemon thread: pool-1121-thread-4 (BLOCKED)
2022-03-18 04:14:50 [43mdestination[0m >  Thread stacktrace: app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:482)
2022-03-18 04:14:50 [43mdestination[0m >         at app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:474)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.lang.Thread.run(Thread.java:833)
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [33mWARN[m i.a.i.b.IntegrationRunner(watchForOrphanThreads):245 - Active non-daemon thread: pool-1121-thread-5 (BLOCKED)
2022-03-18 04:14:50 [43mdestination[0m >  Thread stacktrace: app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:482)
2022-03-18 04:14:50 [43mdestination[0m >         at app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:474)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.lang.Thread.run(Thread.java:833)
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [33mWARN[m i.a.i.b.IntegrationRunner(watchForOrphanThreads):245 - Active non-daemon thread: pool-1121-thread-6 (BLOCKED)
2022-03-18 04:14:50 [43mdestination[0m >  Thread stacktrace: app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:482)
2022-03-18 04:14:50 [43mdestination[0m >         at app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:474)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.lang.Thread.run(Thread.java:833)
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [33mWARN[m i.a.i.b.IntegrationRunner(watchForOrphanThreads):245 - Active non-daemon thread: pool-1121-thread-7 (BLOCKED)
2022-03-18 04:14:50 [43mdestination[0m >  Thread stacktrace: app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:482)
2022-03-18 04:14:50 [43mdestination[0m >         at app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:474)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.lang.Thread.run(Thread.java:833)
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [33mWARN[m i.a.i.b.IntegrationRunner(watchForOrphanThreads):245 - Active non-daemon thread: pool-1121-thread-8 (BLOCKED)
2022-03-18 04:14:50 [43mdestination[0m >  Thread stacktrace: app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:482)
2022-03-18 04:14:50 [43mdestination[0m >         at app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:474)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.lang.Thread.run(Thread.java:833)
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [33mWARN[m i.a.i.b.IntegrationRunner(watchForOrphanThreads):245 - Active non-daemon thread: pool-1121-thread-9 (BLOCKED)
2022-03-18 04:14:50 [43mdestination[0m >  Thread stacktrace: app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:482)
2022-03-18 04:14:50 [43mdestination[0m >         at app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:474)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.lang.Thread.run(Thread.java:833)
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [33mWARN[m i.a.i.b.IntegrationRunner(watchForOrphanThreads):245 - Active non-daemon thread: pool-1121-thread-10 (BLOCKED)
2022-03-18 04:14:50 [43mdestination[0m >  Thread stacktrace: app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:482)
2022-03-18 04:14:50 [43mdestination[0m >         at app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:474)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.lang.Thread.run(Thread.java:833)
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [33mWARN[m i.a.i.b.IntegrationRunner(watchForOrphanThreads):245 - Active non-daemon thread: pool-2080-thread-1 (BLOCKED)
2022-03-18 04:14:50 [43mdestination[0m >  Thread stacktrace: app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:482)
2022-03-18 04:14:50 [43mdestination[0m >         at app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:474)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.lang.Thread.run(Thread.java:833)
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [33mWARN[m i.a.i.b.IntegrationRunner(watchForOrphanThreads):245 - Active non-daemon thread: pool-2080-thread-2 (BLOCKED)
2022-03-18 04:14:50 [43mdestination[0m >  Thread stacktrace: app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:482)
2022-03-18 04:14:50 [43mdestination[0m >         at app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:474)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.lang.Thread.run(Thread.java:833)
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [33mWARN[m i.a.i.b.IntegrationRunner(watchForOrphanThreads):245 - Active non-daemon thread: pool-2080-thread-3 (BLOCKED)
2022-03-18 04:14:50 [43mdestination[0m >  Thread stacktrace: app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:482)
2022-03-18 04:14:50 [43mdestination[0m >         at app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:474)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.lang.Thread.run(Thread.java:833)
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [33mWARN[m i.a.i.b.IntegrationRunner(watchForOrphanThreads):245 - Active non-daemon thread: pool-2080-thread-4 (BLOCKED)
2022-03-18 04:14:50 [43mdestination[0m >  Thread stacktrace: app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:482)
2022-03-18 04:14:50 [43mdestination[0m >         at app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:474)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.lang.Thread.run(Thread.java:833)
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [33mWARN[m i.a.i.b.IntegrationRunner(watchForOrphanThreads):245 - Active non-daemon thread: pool-2080-thread-5 (BLOCKED)
2022-03-18 04:14:50 [43mdestination[0m >  Thread stacktrace: app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:482)
2022-03-18 04:14:50 [43mdestination[0m >         at app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:474)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.lang.Thread.run(Thread.java:833)
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [33mWARN[m i.a.i.b.IntegrationRunner(watchForOrphanThreads):245 - Active non-daemon thread: pool-2080-thread-6 (BLOCKED)
2022-03-18 04:14:50 [43mdestination[0m >  Thread stacktrace: app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:482)
2022-03-18 04:14:50 [43mdestination[0m >         at app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:474)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.lang.Thread.run(Thread.java:833)
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [33mWARN[m i.a.i.b.IntegrationRunner(watchForOrphanThreads):245 - Active non-daemon thread: pool-2080-thread-7 (BLOCKED)
2022-03-18 04:14:50 [43mdestination[0m >  Thread stacktrace: app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:482)
2022-03-18 04:14:50 [43mdestination[0m >         at app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:474)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.lang.Thread.run(Thread.java:833)
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [33mWARN[m i.a.i.b.IntegrationRunner(watchForOrphanThreads):245 - Active non-daemon thread: pool-2080-thread-8 (WAITING)
2022-03-18 04:14:50 [43mdestination[0m >  Thread stacktrace: java.base@17.0.1/jdk.internal.misc.Unsafe.park(Native Method)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.locks.LockSupport.park(LockSupport.java:341)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionNode.block(AbstractQueuedSynchronizer.java:506)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ForkJoinPool.unmanagedBlock(ForkJoinPool.java:3463)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:3434)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:1623)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ArrayBlockingQueue.take(ArrayBlockingQueue.java:420)
2022-03-18 04:14:50 [43mdestination[0m >         at app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:484)
2022-03-18 04:14:50 [43mdestination[0m >         at app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:474)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.lang.Thread.run(Thread.java:833)
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [33mWARN[m i.a.i.b.IntegrationRunner(watchForOrphanThreads):245 - Active non-daemon thread: pool-2080-thread-9 (BLOCKED)
2022-03-18 04:14:50 [43mdestination[0m >  Thread stacktrace: app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:482)
2022-03-18 04:14:50 [43mdestination[0m >         at app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:474)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.lang.Thread.run(Thread.java:833)
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [33mWARN[m i.a.i.b.IntegrationRunner(watchForOrphanThreads):245 - Active non-daemon thread: pool-2080-thread-10 (BLOCKED)
2022-03-18 04:14:50 [43mdestination[0m >  Thread stacktrace: app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:482)
2022-03-18 04:14:50 [43mdestination[0m >         at app//alex.mojaki.s3upload.StreamTransferManager$UploadTask.call(StreamTransferManager.java:474)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-03-18 04:14:50 [43mdestination[0m >         at java.base@17.0.1/java.lang.Thread.run(Thread.java:833)
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [32mINFO[m i.a.i.b.FailureTrackingAirbyteMessageConsumer(close):65 - Airbyte message consumer: succeeded.
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [32mINFO[m i.a.i.d.b.BufferedStreamConsumer(close):217 - executing on success close procedure.
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [32mINFO[m i.a.i.d.b.BufferedStreamConsumer(flushQueueToDestination):181 - Flushing buffer: 9316469 bytes
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [32mINFO[m i.a.i.d.b.BufferedStreamConsumer(lambda$flushQueueToDestination$1):185 - Flushing contacts: 8975 records
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [32mINFO[m i.a.i.d.j.c.s.S3StreamCopier(prepareStagingFile):95 - S3 upload part size: 10 MB
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [32mINFO[m i.a.i.d.s.c.S3CsvWriter(<init>):58 - Full S3 path for stream 'contacts': s3://textus-airbyte-workspace/public/contacts/2022_03_17_1647550729493_a7130562-dba7-42c9-b0a3-51e62623ff9e.csv
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [32mINFO[m i.a.i.d.s.u.S3StreamTransferManagerHelper(getDefault):55 - PartSize arg is set to 10 MB
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [32mINFO[m a.m.s.StreamTransferManager(getMultiPartOutputStreams):329 - Initiated multipart upload to textus-airbyte-workspace/public/contacts/2022_03_17_1647550729493_a7130562-dba7-42c9-b0a3-51e62623ff9e.csv with full ID dMqLPPEM28MPI0qPqoZkWACPrixnl4nrPYSExzfux2w56GXT0.iLDQoGWgjtcD7i7obQNv92dw7l1.b8AWpKDJAXXIcY9ATdbHd8irK9eCLHVqBitiPF8oqk7QQo2bOl
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [32mINFO[m i.a.i.d.s.w.BaseS3Writer(close):113 - Uploading remaining data for stream 'contacts'.
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [32mINFO[m a.m.s.MultiPartOutputStream(close):158 - Called close() on [MultipartOutputStream for parts 1 - 10000]
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [32mINFO[m a.m.s.MultiPartOutputStream(close):158 - Called close() on [MultipartOutputStream for parts 1 - 10000]
2022-03-18 04:14:50 [43mdestination[0m > 2022-03-18 04:14:50 [33mWARN[m a.m.s.MultiPartOutputStream(close):160 - [MultipartOutputStream for parts 1 - 10000] is already closed

```

</details>

## Steps to Reproduce
1. Have tables with 100's of millions of rows
2. Upgrade Airbyte from `0.35.13` to `0.35.55`
3. Upgrade all connectors to their latest
4. Attempt to reset and ReSync all data from source

## Are you willing to submit a PR?

If I know the language and what to do sure.

**Comment by qops1981** [comment]
I was able to reproduce the issues with a table that only have about 2 mil rows using the original config. I will test the downgrade of the Snowflake connector now on that table since it take less than and hour now.

**Comment by qops1981** [comment]
Downgrading the Snowflake connector to `0.4.7` worked. I am going to incrementally bump and test that connector until I run into the issue again. I will post where it start here when I find it.

**Comment by qops1981** [comment]
I was able to confirm up to `0.4.12` worked without the issue. I was not able to load `0.4.13` or `0.4.14` as the UI said "something went wrong" ( tried several times ), but I was able to load `0.4.15` and that was the start of our issue. We will hold at `0.4.12` for the time being.

**Comment by qops1981** [comment]
Upgrading Airbyte to `0.35.57` I was able to load the `0.4.13` Snowflake connector and was again able to reproduce the issue. I was able to reproduce this issue several times with regards to this connector. I am now confident that the issue starts with `0.4.13` Snowflake connector.

**Comment by qops1981** [comment]
Note: on the "Critical" state of this bug; this can be dropped down as far as we are concerned. I am still testing with the `0.4.12` Snowflake connector, but provided all goes well, we will be back up and running.

**Comment by marcosmarxm** [comment]
@qops1981 I'm closing due inactivity and corresponds to an old version of Airbyte. Please open a new issue if the problem persists.
